pkg/cmd/cli/install/install.go

```diff
@@ -0,0 +1,204 @@
+	if err != nil {
+		return nil, err
+	}
+	return &install.VeleroOptions{
+		Namespace:    o.Namespace,
+		Image:        o.Image,
+		ProviderName: o.ProviderName,
+		Bucket:       o.BucketName,
+		Prefix:       o.Prefix,
+		SecretData:   secretData,
+		RestoreOnly:  o.RestoreOnly,
+		UseRestic:    o.UseRestic,
+		BSLConfig:    o.BackupStorageConfig.Data(),
+		VSLConfig:    o.VolumeSnapshotConfig.Data(),
+	}, nil
+}
+
+// NewCommand creates a cobra command.
+func NewCommand(f client.Factory) *cobra.Command {
+	o := NewInstallOptions()
+	c := &cobra.Command{
+		Use:   "install",
+		Short: "Install Velero",
+		Long: `
+Install Velero onto a Kubernetes cluster using the supplied provider information, such as
+the provider's name, a bucket name, and a file containing the credentials to access that bucket.
+A prefix within the bucket and configuration for the backup store location may also be supplied.
+Additionally, volume snapshot information for the same provider may be supplied.
+
+All required CustomResourceDefinitions will be installed to the server, as well as the
+Velero Deployment and associated Restic DaemonSet.
+
+The provided secret data will be created in a Secret named 'cloud-credentials'.
+
+All namespaced resources will be placed in the 'velero' namespace.
+
+Use '-o yaml' or '-o json'  with '--dry-run' to output all generated resources as text instead of sending the resources to the server.
+This is useful as a starting point for more customized installations.
+		`,
+		Example: `	# velero install --bucket mybucket --provider gcp --secret-file ./gcp-service-account.json
+
+	# velero install --bucket backups --provider aws --secret-file ./aws-iam-creds --backup-location-config region=us-east2 --snapshot-location-config region=us-east2
+
+	# velero install --bucket backups --provider aws --secret-file ./aws-iam-creds --backup-location-config region=us-east2 --snapshot-location-config region=us-east2 --use-restic
+		`,
+		Run: func(c *cobra.Command, args []string) {
+			cmd.CheckError(o.Validate(c, args, f))
+			cmd.CheckError(o.Complete(args, f))
+			cmd.CheckError(o.Run(c))
+		},
+	}
+
+	o.BindFlags(c.Flags())
+	output.BindFlags(c.Flags())
+	output.ClearOutputFlagDefault(c)
+
+	return c
+}
+
+// Run executes a command in the context of the provided arguments.
+func (o *InstallOptions) Run(c *cobra.Command) error {
+	vo, err := o.AsVeleroOptions()
+	if err != nil {
+		return err
+	}
+
+	resources, err := install.AllResources(vo)
+	if err != nil {
+		return err
+	}
+
+	if _, err := output.PrintWithFormat(c, resources); err != nil {
+		return err
+	}
+
+	if o.DryRun {
+		return nil
+	}
+
+	clientConfig, err := client.Config("", "", fmt.Sprintf("%s-%s", c.Parent().Name(), c.Name()))
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := dynamic.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	err = install.Install(client.NewDynamicFactory(dynamicClient), resources, os.Stdout)
```

Comment:
I'm wondering if it would make sense to expose a `--wait` flag, and only to wait for the Deployment (and possibly DaemonSet, see below) to become ready if that's true. WDYT?

Reply:
I'm thinking that might be a good idea, too. It simplifies the actual install logic. I still think we need the wait for the CRDs, though.

Thinking about usage, if the user doesn't provide `--wait` and there's an issue, how would they know? They'll have to do `kubectl logs deploy/velero -n velero` and/or `kubectl logs ds/restic -n velero` to check it. Maybe we output those commands on completion, kind of like we do with `backup create`.

Reply:
Yeah, agree we need to wait for the CRDs regardless, since not doing that could cause creation of BSL/VSL instances to fail.

Yeah -- I think outputting the relevant `kubectl get` commands and `kubectl logs` commands would make sense there, and we could output some text saying that the API objects have been created, but we haven't verified they're up and running.